backend/src/main/java/com/bakdata/conquery/util/search/TrieSearch.java

```diff
@@ -119,80 +147,88 @@ private Stream<String> split(String keyword) {
 	}
 
 	/**
-	 * A lower weight implies more relevant words.
+	 * Calculate and update weights for all queried items
 	 */
-	private double weightWord(String keyword, String itemWord) {
+	private void updateWeights(String query, final Map<String, List<T>> items, Object2LongMap<T> itemWeights, boolean original) {
+		if (items == null) {
+			return;
+		}
+
+		for (Map.Entry<String, List<T>> entry : items.entrySet()) {
+			final List<T> hits = entry.getValue();
+
+			if (hits.isEmpty()) {
+				continue;
+			}
+
+			long weight = weightWord(query, entry.getKey(), original);
 
-		// Test if the word is an original word and not a suffix.
-		final boolean isOriginal = isOriginal(itemWord);
+			// We combine hits additively to favor items with multiple hits
+			for (T item : hits) {
+				itemWeights.put(item, itemWeights.getOrDefault(item, 1) + weight);
+			}
+		}
+	}
 
-		final double keywordLength = keyword.length();
-		final double itemLength = itemWord.length() - (isOriginal ? 1 : 0);
+	/**
+	 * Returns an empty stream if word is shorter than ngramLength or ngramLength is Integer.MAX_VALUE
+	 */
+	public Stream<String> ngramSplit(String word) {
 
-		// keyword is prefix of itemWord
-		assert itemLength >= keywordLength;
+		// Any String  is its own ngram when ngramLength is Integer.MAX_VALUE.
+		if (word.length() < ngramLength || ngramLength == Integer.MAX_VALUE) {
```

Comment:
```suggestion
		if (word.length() < ngramLength) {
```

bisschen tautologisch oder? :D